# Can you find a better way to segment your customers?

## 📖 Background
You work for a medical device manufacturer in Switzerland. Your company manufactures orthopedic devices and sells them worldwide. The company sells directly to individual doctors who use them on rehabilitation and physical therapy patients.

Historically, the sales and customer support departments have grouped doctors by geography. However, the region is not a good predictor of the number of purchases a doctor will make or their support needs.

Your team wants to use a data-centric approach to segmenting doctors to improve marketing, customer service, and product planning. 

## 💾 The data

The company stores the information you need in the following four tables. Some of the fields are anonymized to comply with privacy regulations.

#### Doctors contains information on doctors. Each row represents one doctor.
- "DoctorID" - is a unique identifier for each doctor.
- "Region" - the current geographical region of the doctor.
- "Category" - the type of doctor, either 'Specialist' or 'General Practitioner.'
- "Rank" - is an internal ranking system. It is an ordered variable: The highest level is Ambassadors, followed by Titanium Plus, Titanium, Platinum Plus, Platinum, Gold Plus, Gold, Silver Plus, and the lowest level is Silver.
- "Incidence rate"  and "R rate" - relate to the amount of re-work each doctor generates.
- "Satisfaction" - measures doctors' satisfaction with the company.
- "Experience" - relates to the doctor's experience with the company.
- "Purchases" - purchases over the last year.

#### Orders contains details on orders. Each row represents one order; a doctor can place multiple orders.
- "DoctorID" - doctor id (matches the other tables).
- "OrderID" - order identifier.
- "OrderNum" - order number.
- "Conditions A through J" - map the different settings of the devices in each order. Each order goes to an individual patient.

#### Complaints collects information on doctor complaints.
- "DoctorID" - doctor id (matches the other tables).
- "Complaint Type" - the company's classification of the complaints.
- "Qty" - number of complaints per complaint type per doctor.

#### Instructions has information on whether the doctor includes special instructions on their orders.
- "DoctorID" - doctor id (matches the other tables).
- "Instructions" - 'Yes' when the doctor includes special instructions, 'No' when they do not.

In [1]:
import pandas as pd
doctors = pd.read_csv('data/doctors.csv')
doctors

DoctorID   Region              Category  ... Satisfaction  Experience  Purchases
0     AHDCBA     4 15            Specialist  ...        53.85        1.20       49.0
1     ABHAHF   1 8 T4  General Practitioner  ...       100.00        0.00       38.0
2      FDHFJ   1 9 T4            Specialist  ...           --        0.00       34.0
3     BJJHCA  1 10 T3            Specialist  ...           --        0.48       29.0
4      FJBEA  1 14 T4            Specialist  ...        76.79        0.75       24.0
..       ...      ...                   ...  ...          ...         ...        ...
432   AIABDJ     1 10            Specialist  ...        11.76        0.77       35.0
433   BBAJCF   1 9 T4            Specialist  ...           --        0.11       19.0
434    GGCFB  1 19 T4            Specialist  ...           --        0.27       22.0
435    FDCEG      1 9            Specialist  ...       100.00        0.32       25.0
436    EIEIB     1 13            Specialist  ...        11.11        0.39       64.0

[437 rows x 9 columns]

In [2]:
orders = pd.read_csv('data/orders.csv')
orders

DoctorID  OrderID OrderNum  Condition A  Condition B Condition C  \
0     ABJEAI  DGEJFDC  AIBEHCJ        False        False       False   
1      HBIEA  DGAJDAH  AIJIHGB        False         True         NaN   
2      GGCCD  DGBBDCB  AFEIHFB        False        False       False   
3      EHHGF  DGCDCCF  AIBJJEE        False        False       False   
4      EHHGF  DGCFAGC  AEDBBDC        False        False       False   
..       ...      ...      ...          ...          ...         ...   
252    FJFEG  DGDBGJH  AIBCDCE        False         True       False   
253    FJFEG  DGDBGGI  AIBCDGD        False        False        True   
254    FJFEG  DGEABGI  AIBFJHE         True        False       False   
255    FJFEG  DGEABGI  AIBFJHE         True        False       False   
256   AIABDJ  DFICBFD  AEFABFI        False        False       False   

     Condition D Condition F Condition G  Condition H  Condition I Condition J  
0          False       False        True         True        False      Before  
1          False       False        True        False         True      Before  
2          False       False       False        False        False         NaN  
3           True       False       False         True        False      Before  
4          False       False       False        False        False         NaN  
..           ...         ...         ...          ...          ...         ...  
252        False       False        True         True        False      Before  
253        False       False       False         True        False      Before  
254        False       False       False        False        False      Before  
255         True       False       False        False        False      Before  
256        False       False       False        False        False         NaN  

[257 rows x 12 columns]

In [3]:
complaints = pd.read_csv('data/complaints.csv')
complaints

DoctorID Complaint Type  Qty
0      EHAHI        Correct   10
1      EHDGF        Correct    2
2      EHDGF        Unknown    3
3      EHDIJ        Correct    8
4      EHDIJ      Incorrect    2
..       ...            ...  ...
430   BHGIFC      Incorrect    1
431   BHHDDF        Correct    1
432   CJAFAB      Incorrect    1
433   CAAHID        Correct    2
434   CAECFG        Correct    2

[435 rows x 3 columns]

In [4]:
instructions = pd.read_csv('data/instructions.csv')
instructions

DoctorID Instructions
0    ADIFBD          Yes
1    ABHBED           No
2     FJFEG          Yes
3    AEBDAB           No
4    AJCBFE          Yes
..      ...          ...
72   ABEAFF          Yes
73    FCGCI          Yes
74    FBAHD          Yes
75    FCABB          Yes
76    GHDFB          Yes

[77 rows x 2 columns]

## 💪 Competition challenge

Create a report that covers the following:
1. How many doctors are there in each region? What is the average number of purchases per region?
2. Can you find a relationship between purchases and complaints?
3. Define new doctor segments that help the company improve marketing efforts and customer service.
4. Identify which features impact the new segmentation strategy the most.
5. Your team will need to explain the new segments to the rest of the company. Describe which characteristics distinguish the newly defined segments.

## 🧑‍⚖️ Judging criteria

| CATEGORY | WEIGHTING | DETAILS                                                              |
|:---------|:----------|:---------------------------------------------------------------------|
| **Recommendations** | 35%       | <ul><li>Clarity of recommendations - how clear and well presented the recommendation is.</li><li>Quality of recommendations - are appropriate analytical techniques used & are the conclusions valid?</li><li>Number of relevant insights found for the target audience.</li></ul>       |
| **Storytelling**  | 35%       | <ul><li>How well the data and insights are connected to the recommendation.</li><li>How the narrative and whole report connects together.</li><li>Balancing making the report in-depth enough but also concise.</li></ul> |
| **Visualizations** | 20% | <ul><li>Appropriateness of visualization used.</li><li>Clarity of insight from visualization.</li></ul> |
| **Votes** | 10% | <ul><li>Up voting - most upvoted entries get the most points.</li></ul> |

## ✅ Checklist before publishing into the competition
- Rename your workspace to make it descriptive of your work. N.B. you should leave the notebook name as notebook.ipynb.
- **Remove redundant cells** like the judging criteria, so the workbook is focused on your story.
- Make sure the workbook reads well and explains how you found your insights. 
- Try to include an **executive summary** of your recommendations at the beginning.
- Check that all the cells run without error

## ⌛️ Time is ticking. Good luck!